File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/javascript_solutions/892-surface-area-of-3d-shapes.js
var surfaceArea = function(grid) {
    let area = 0;
    const n = grid.length;
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            const height = grid[i][j];
            if (height > 0) {
                area += 2;
                area += 4 * height;
                if (i > 0) area -= 2 * Math.min(height, grid[i - 1][j]);
                if (j > 0) area -= 2 * Math.min(height, grid[i][j - 1]);
            }
        }
    }
    return area;
};

File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/rust_solutions/892-surface-area-of-3d-shapes.rs
impl Solution {
    pub fn surface_area(grid: Vec<Vec<i32>>) -> i32 {
        let n = grid.len();
        let m = grid[0].len();
        let mut total_surface_area = 0;

        for i in 0..n {
            for j in 0..m {
                let h = grid[i][j];

                if h > 0 {
                    total_surface_area += 2; // Top and bottom surfaces
                    total_surface_area += 4 * h; // Initial side surfaces

                    // Subtract overlapping surfaces with neighbors

                    // Check right neighbor
                    if j + 1 < m {
                        let h_right = grid[i][j + 1];
                        total_surface_area -= 2 * std::cmp::min(h, h_right);
                    }

                    // Check down neighbor
                    if i + 1 < n {
                        let h_down = grid[i + 1][j];
                        total_surface_area -= 2 * std::cmp::min(h, h_down);
                    }
                }
            }
        }

        total_surface_area
    }
}